subsys/console/getchar.c

```diff
@@ -117,3 +120,25 @@ void console_init(void)
 	uart_irq_callback_set(uart_dev, uart_isr);
 	uart_irq_rx_enable(uart_dev);
 }
+
+#elif defined(CONFIG_SOC_POSIX)
+
+#include <stdio.h>
+
+int console_putchar(char c)
+{
+	putchar(c);
+}
+
+u8_t console_getchar(void)
+{
+	return getchar();
+}
+
+void console_init(void)
+{
+	setvbuf(stdin, NULL, _IONBF, 0);
```

Comment:
This conflicts directly with the native_posix stdout configuration.

Reply:
Yep, that's what I meant. Can you please wrap your code in `#ifdef CONFIG_CONSOLE_PULL`?

Reply:
you mean the stdin part? Would it not be better to do that in the configuration files that select NATIVE_POSIX_STDIN_CONSOLE?
If you mean for the sdout part, that would disconnect printk() which would not be too nice..

Reply:
In my list, it would be (almost) all code in native_posix_console.c. We effectively have mutual exclusion - CONFIG_CONSOLE_PULL is off by default, but if it's on, a user requesting it knows/accepts the fact that they lose access to native Zephyr console.

And all that are just my ideas how to deal with it, the situation is definitely "interim" (if not say "not clean"), as I described, and I just wanted to show and describe it via some code (vs e.g. just reporting it to you). So, if you have better ideas, they're appreciated.

Reply:
> that they lose access to native Zephyr console.

Aren't the subsystem/console sample apps themselves are using `printk()`?

Reply:
> Aren't the subsystem/console sample apps themselves using printk()?

Right, so #ifdef scope apparently indeed should be scaled down. Well, ok, I guess I'll need to wait why your PR will be merged, then experiment with how to make CONSOLE_PULL play well with it, then submit combined proposal patch.